edgedb-tokio/src/tutorial.md

```diff
@@ -0,0 +1,475 @@
+
+```rust
+let query = "select <int32>$0";
+let argument = 9i16; // Rust client will expect an int16
+let query_res: Result<Value, _> = client.query_required_single(query, &(argument,)).await;
+assert!(query_res
+    .unwrap_err()
+    .to_string()
+    .contains("expected std::int16"));
+```
+
+## Using JSON
+
+EdgeDB can cast any type to JSON with `<json>`, but the `_json` methods don't require this cast in the query. This result can be turned into a String and used to respond to some JSON API request directly, unpacked into a struct using `serde` and `serde_json`, etc.
+
+```rust
+#[derive(Debug, Deserialize)]
+pub struct Account {
+    pub username: String,
+    pub id: Uuid,
+}
+
+// No need for <json> cast here
+let query = "select Account { 
+      username,
+      id
+    } filter .username = <str>$0;";
+
+// Assuming we know there will only be one result we can use query_single_json;
+// otherwise query_json which returns a map of json
+let json_res = client
+    .query_single_json(query, &("SomeUserName",))
+    .await?
+    .unwrap();
+
+// Format: {"username" : "SomeUser1", "id" : "7093944a-fd3a-11ed-a013-c7de12ffe7a9"}
+let as_string = json_res.to_string();
+let as_account: Account = serde_json::from_str(&json_res)?;
+```
+
+## Execute
+
+The `execute` method was added in June 2023 and will show up inside the next version of the `edgedb-tokio` crate. In the meantime it can be used by setting your edgedb dependencies to the git repo:
+
+```
+edgedb-derive = { git = "https://github.com/edgedb/edgedb-rust" }
+edgedb-tokio = { git = "https://github.com/edgedb/edgedb-rust" }
+edgedb-protocol = { git = "https://github.com/edgedb/edgedb-rust" }
+```
+
+`execute` doesn't return anything (a successful execute returns an `Ok(())`) which is convenient for things like updates or commands where we don't care about getting an output if it works:
+
+```rust
+client.execute("update Account set {username := .username ++ '!'};", &()).await?;
+client.execute("create superuser role project;", &()).await?;
+client.execute("alter role project set password := 'STRONGpassword';", &()).await?;
+
+// Returns Ok(()) upon success but error info will be returned of course
+let command = client.execute("create type MyType {};", &()).await;
+assert!(command.unwrap_err().to_string().contains("bare DDL statements are not allowed"));
+```
+
+## Transactions
+
+The client also has a `.transaction()` method that allows atomic [transactions](https://www.edgedb.com/docs/edgeql/transactions). 
+
+Note: What often may seem to require an atomic transaction can instead be achieved with links and [backlinks](https://www.edgedb.com/docs/edgeql/paths#backlinks) which are both idiomatic and easy to use in EdgeDB. For example, if one object holds a `required link` to two other objects and each of these two objects has a single banklink to the first one, simply inserting the first object will effectively change the state of the other two instantaneously.
```

Comment:
```suggestion
Note: What often may seem to require an atomic transaction can instead be achieved with links and [backlinks](https://www.edgedb.com/docs/edgeql/paths#backlinks) which are both idiomatic and easy to use in EdgeDB. For example, if one object holds a `required link` to two other objects and each of these two objects has a single backlink to the first one, simply inserting the first object will effectively change the state of the other two instantaneously.
```

In addition to that suggestion, I have trouble following this.

> …simply inserting the first object will effectively change the state of the other two instantaneously.

If two objects have a link to an object, that would dictate that the object has already been inserted, so inserting it would just fail. Should this be "updating" instead of "inserting?"